lib/graphql/subscriptions/event.rb

```diff
@@ -65,12 +65,24 @@ def stringify_args(arg_owner, args)
 
               if arg_defn
                 normalized_arg_name = camelized_arg_name
-                next_args[normalized_arg_name] = stringify_args(arg_defn.type, v)
               else
                 normalized_arg_name = arg_name
                 arg_defn = get_arg_definition(arg_owner, normalized_arg_name)
               end
-              next_args[normalized_arg_name] = stringify_args(arg_defn.type, v)
+              arg_base_type = arg_defn.type
+              while arg_base_type.respond_to?(:of_type)
+                arg_base_type = arg_base_type.of_type
+              end
+              # In the case where the value being emitted is seen as a "JSON"
+              # type, treat the value as one atomic unit of serialization
+              is_json_definition = arg_base_type && arg_base_type <= GraphQL::Types::JSON
+              if is_json_definition
+                next_args[normalized_arg_name] = v.respond_to?(:to_json) ? v.to_json : v
```

Comment:
Usually, this method sorts hashes to make sure that clients who provide arguments _in different order_ still end up subscribing to the same topic. Should we also consider ordering here? 

In any case, how about adding a test to demonstrate the behavior with Types::JSON and ordering, so at least we can _know_ how it works?